apps/src/face_detection/openni_face_detection.cpp

```diff
@@ -5,85 +5,97 @@
+    pcl::visualization::PointCloudColorHandlerRGBField<OpenNIFrameSource::PointT>
+        handler_keypoints(scene_vis);
+    vis.addPointCloud<OpenNIFrameSource::PointT>(
+        scene_vis, handler_keypoints, "scene_cloud");
+
+    if (heat_map) {
+      pcl::PointCloud<pcl::PointXYZI>::Ptr intensity_cloud(
+          new pcl::PointCloud<pcl::PointXYZI>);
+      fdrf.getFaceHeatMap(intensity_cloud);
+      pcl::visualization::PointCloudColorHandlerGenericField<pcl::PointXYZI>
+          handler_keypoints(intensity_cloud, "intensity");
+      vis.addPointCloud<pcl::PointXYZI>(intensity_cloud, handler_keypoints, "heat_map");
     }
 
-    if (show_votes)
-    {
-      //display votes_
-      pcl::PointCloud<pcl::PointXYZ>::Ptr votes_cloud (new pcl::PointCloud<pcl::PointXYZ> ());
-      fdrf.getVotes (votes_cloud);
-      pcl::visualization::PointCloudColorHandlerCustom < pcl::PointXYZ > handler_votes (votes_cloud, 255, 0, 0);
-      vis.addPointCloud < pcl::PointXYZ > (votes_cloud, handler_votes, "votes_cloud");
-      vis.setPointCloudRenderingProperties (pcl::visualization::PCL_VISUALIZER_POINT_SIZE, 14, "votes_cloud");
-      vis.setPointCloudRenderingProperties (pcl::visualization::PCL_VISUALIZER_OPACITY, 0.5, "votes_cloud");
-      vis.setPointCloudRenderingProperties (pcl::visualization::PCL_VISUALIZER_OPACITY, 0.75, "votes_cloud");
+    if (show_votes) {
+      // display votes_
+      pcl::PointCloud<pcl::PointXYZ>::Ptr votes_cloud(
+          new pcl::PointCloud<pcl::PointXYZ>());
+      fdrf.getVotes(votes_cloud);
+      pcl::visualization::PointCloudColorHandlerCustom<pcl::PointXYZ> handler_votes(
+          votes_cloud, 255, 0, 0);
+      vis.addPointCloud<pcl::PointXYZ>(votes_cloud, handler_votes, "votes_cloud");
+      vis.setPointCloudRenderingProperties(
+          pcl::visualization::PCL_VISUALIZER_POINT_SIZE, 14, "votes_cloud");
+      vis.setPointCloudRenderingProperties(
+          pcl::visualization::PCL_VISUALIZER_OPACITY, 0.5, "votes_cloud");
+      vis.setPointCloudRenderingProperties(
+          pcl::visualization::PCL_VISUALIZER_OPACITY, 0.75, "votes_cloud");
     }
 
     std::vector<Eigen::VectorXf> heads;
-    fdrf.getDetectedFaces (heads);
+    fdrf.getDetectedFaces(heads);
 
-    face_detection_apps_utils::displayHeads (heads, vis);
+    face_detection_apps_utils::displayHeads(heads, vis);
 
-    vis.setRepresentationToSurfaceForAllActors ();
-    vis.spinOnce ();
+    vis.setRepresentationToSurfaceForAllActors();
+    vis.spinOnce();
 
-    vis.removeAllPointClouds ();
-    vis.removeAllShapes ();
+    vis.removeAllPointClouds();
+    vis.removeAllShapes();
   }
 }
 
-//./bin/pcl_openni_face_detector -face_threshold 0.99 -max_variance 2400 -min_votes_size 300 -stride_sw 4 -heat_map 0 -show_votes 1 -pose_refinement 1 -icp_iterations 5 -model_path face_model.pcd -forest_fn forest.txt
-int main(int argc, char ** argv)
+//./bin/pcl_openni_face_detector -face_threshold 0.99 -max_variance 2400 -min_votes_size
```

Comment:
What to do here?

Reply:
I can't find a `face_model.pcd` in the Git repo. So I think we can remove this example call, as you can already see some lines below which parameter exists and I guess the parameter are specific to the model.